### PR TITLE
adding support for type generation from schemas

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/1201614831475344/1205190929920546/f

image

### CODE REVIEW

**Comment by jonathanKingston** [comment]
@shakyShane let me know when you're ready for review here, we've still got the failing tests.

**Comment by shakyShane** [comment]
@jonathanKingston ready now - had a windows thing to fix, but all sorted now :)

File: scripts/build-types.mjs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+export async function buildTypes(mapping = defaultMapping) {
+    for (let [featureName, manifest] of Object.entries(mapping)) {
+        const typescript = await compileFromFile(manifest.schema, {
+            bannerComment: `
+            // eslint-disable
+            /**
+             * @module ${featureName} Schema
+             * @description 
+             * These types are auto-generated from schema files within https://github.com/duckduckgo/privacy-configuration
```
Comment (jonathanKingston) [inline scripts/build-types.mjs:36]:
```suggestion
             * These types are auto-generated from schema files.
             * scripts/build-types.mjs is responsible for type generation.
             * See the privacy-configuration repo for the schema files:
             * https://github.com/duckduckgo/privacy-configuration
```

Comment (jonathanKingston) [reply scripts/build-types.mjs:36]: Feel free to take this or leave it, it just makes the lines wrap a little less and is a little clearer where it's made from.

If you're taking it generate the type again :)

**Review by jonathanKingston — DISMISSED** [review]

**Review by jonathanKingston — APPROVED** [review]
Great stuff thanks for doing this!

**Pull request merged**